neo/rawio/medrawio.py

```diff
@@ -0,0 +1,238 @@
+        
+        # set the matrix calls to be "sample_major" as opposed to "channel_major"
+        self.sess.set_major_dimension("sample")
+        
+        # Find highest sampling frequency, and use this for all channels
+        self.highest_sampling_freq = None
+        self.reference_channel = None
+        self.num_samples_in_session = 0
+        self.num_channels_in_session = len(self.sess.session_info['channels'])
+        for c in range (self.num_channels_in_session):
+            if self.highest_sampling_freq is None:
+                self.highest_sampling_freq = self.sess.session_info['channels'][c]['metadata']['sampling_frequency']
+                self.reference_channel = self.sess.session_info['channels'][c]['metadata']['channel_name']
+            if self.sess.session_info['channels'][c]['metadata']['sampling_frequency'] > self.highest_sampling_freq:
+                self.highest_sampling_freq = self.sess.session_info['channels'][c]['metadata']['sampling_frequency']
+                self.reference_channel = self.sess.session_info['channels'][c]['metadata']['channel_name']
+                    
+        # Using the reference channel, find out how many samples are in the session
+        # This reference channel will also be used later in the get_matrix_by_sample() function
+        # to retrieve samples from the MED session.
+        self.sess.set_reference_channel(self.reference_channel)
+        self.num_samples_in_session = self.sess.get_globals_number_of_session_samples()
+               
+        # Set session start/end time boundaries
+        self.session_start_time = 0
+        self.session_end_time = (self.num_samples_in_session / self.highest_sampling_freq) * 1e6
+        
+        # create signal streams information
+        signal_streams = []
+        for c in range(1):
+            name = f'stream ({self.highest_sampling_freq} Hz)'
+            stream_id = c
+            signal_streams.append((name, stream_id))
+        signal_streams = np.array(signal_streams, dtype=_signal_stream_dtype)
+
+        signal_channels = []
+        for c in range(self.num_channels_in_session):
+            ch_name = self.sess.session_info['channels'][c]['metadata']['channel_name']
+            chan_id = self.sess.session_info['channels'][c]['metadata']['acquisition_channel_number']
+            sr = self.highest_sampling_freq
+            dtype = 'float64'
+            units = 'uV'
+            gain = 1
+            offset = 0.
+            stread_id = 0
+            signal_channels.append((ch_name, chan_id, sr, dtype, units, gain, offset, stream_id))
+    
+        signal_channels = np.array(signal_channels, dtype=_signal_channel_dtype)
+
+        # no unit/epoch information contained in MED
+        spike_channels = []
+        spike_channels = np.array(spike_channels, dtype=_spike_channel_dtype)
+        
+        # Events
+        # events are not being read by this Neo wrapper
```

Comment:
How complicated would it be to add events here? Depending on the data these might be essential for preprocessing or analysis...